swarm/network/discovery_test.go

```diff
@@ -58,3 +67,154 @@ func TestDiscovery(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+// TestSubpeersMsg tests that the correct set of peers is suggested
+// to another peer with a given depth and kademlia
+func TestSubpeersMsg(t *testing.T) {
+
+	// This is the defined depth
+	testDepth := rand.Intn(4) + 1
+
+	// construct ProtocolTester and hive
+	params := NewHiveParams()
+	// setup
+	prvkey, err := crypto.GenerateKey()
+	if err != nil {
+		t.Fatal(err)
+	}
+	addr := PrivateKeyToBzzKey(prvkey)
+	to := NewKademlia(addr, NewKadParams())
+	hive := NewHive(params, to, nil) // hive
+
+	s, err := newBzzBaseTester(t, 1, prvkey, DiscoverySpec, hive.Run)
```

Comment:
this needs to be after you fill hive/kademlia, otherwise there is a race at which point pivot sends  out `subPeersMsg`. we had this right already twice